app/common/data/models.py

```diff
@@ -244,20 +244,23 @@ class Form(BaseModel):
     @property
     def questions(self) -> list["Question"]:
         """Consistently returns all questions in the form, respecting order and any level of nesting."""
-        return get_ordered_nested_questions_for_components(self.components)
+        return [q for q in get_ordered_nested_components(self.components) if isinstance(q, Question)]
+
+    @property
+    def all_components(self) -> list["Component"]:
+        return get_ordered_nested_components(self.components)
 
 
 # todo: unit test reasonably extensively
```

Comment:
I feel like you've paid this down now, feel free to reap the rewards and drop the todo 🙇